TypeScript/HW_11/task 11.2.3.ts

```diff
@@ -0,0 +1,19 @@
+// 3. Реализуйте функцию filter(), которая принимает на вход массив чисел и предикат (коллбэк), 
+//     который будет использоваться для проверки каждого числа на соответствие требованиям. 
+//     Помимо самой функции следует реализовать алиасы типов для функций и аттрибутов. 
+//     Пример функции:
+//     const numbers = [1, -5, 2, 3, 4, 133];
+//     filter(numbers, (n) => n > 3); // [4, 133]
+//     filter(numbers, (n) => n % 2 == 0); // [2, 4]
+//     Параметры функции: Массив чисел и Анонимная функция, принимающая на вход число и возвращающая логическое значение.
+
+type ArrNum = number[];
+
+const numbers: ArrNum = [1, -5, 2, 3, 4, 133];
+
+function filter(array: ArrNum, callback: (n: number) => boolean) {
+    return array.filter(callback);
```

Comment:
Try not to use array method
Implementation with `for` loop looks not bad

Reply:
Thanks for clarification. Fixed